google/gax/api_callable.py

```diff
@@ -272,18 +280,54 @@ def construct_settings(
         default config and config_override will be specified by users.
       retry_names: A dictionary mapping the strings referring to response status
         codes to the Python objects representing those codes.
+      metrics_headers: Dictionary of headers to be passed for analytics.
+        Sent as a dictionary; eventually becomes a space-separated string
+        (e.g. 'foo/1.0.0 bar/3.14.1').
       kwargs: The keyword arguments to be passed to the API calls.
 
     Raises:
       KeyError: If the configuration for the service in question cannot be
         located in the provided ``client_config``.
     """
     # pylint: disable=too-many-locals
+    # pylint: disable=protected-access
     defaults = {}
     bundle_descriptors = bundle_descriptors or {}
+    metrics_headers = collections.OrderedDict(metrics_headers)
```

Comment:
So just confirming: this means we want any user-provided custom headers to come first?